player-avatar-binding.js

```diff
@@ -9,12 +9,16 @@ const avatarSymbol = 'avatar'; // Symbol('avatar');
 
 export function applyPlayerTransformsToAvatar(player, session, rig) {
   if (!session) {
-    rig.inputs.hmd.position.copy(player.avatarBinding.position);
-    rig.inputs.hmd.quaternion.copy(player.avatarBinding.quaternion);
-    rig.inputs.leftGamepad.position.copy(player.leftHand.position);
-    rig.inputs.leftGamepad.quaternion.copy(player.leftHand.quaternion);
-    rig.inputs.rightGamepad.position.copy(player.rightHand.position);
-    rig.inputs.rightGamepad.quaternion.copy(player.rightHand.quaternion);
+    rig.inputs.hmd.matrix.setPosition(player.avatarBinding.position);

```

Comment:
Imo this would be better with `.compose`, or setting the positions/quats/scales and then `updateMatrixWorld(true)`.

In fact I think this has a bug, because `makeRotationFromQuaternion` completely overwirtes the matrix: https://github.com/mrdoob/three.js/blob/dev/src/math/Matrix4.js#L289

Reply:
good point

Reply:
ill rewrite this

Reply:
>or setting the positions/quats/scales and then updateMatrixWorld(true).

Does not work with THREE.Object3D.DefaultMatrixAutoUpdate  set to false